pkg/egressgateway/manager.go

```diff
@@ -630,21 +630,27 @@ nextIpRule:
 		}
 	}
 
-	// Then go through each interface on the node
-	links, err := netlink.LinkList()
+	// Fetch all IP routes, and delete the unused EgressGW-specific routes:
+	routes, err := netlink.RouteList(nil, netlink.FAMILY_V4)
```

Comment:
Hey, heads up that after https://github.com/cilium/cilium/pull/24677 is merged this can be done more efficiently by querying the new in-memory route table.